src/Algebra/Graph/Relation/Symmetric/Internal.hs

```diff
@@ -0,0 +1,193 @@
+                                                      (domain x   `Set.union` domain y)
+                                                      (relation x `Set.union` relation y)
+
+-- | /Connect/ two graphs. This is an associative operation with the identity
+-- 'empty', which distributes over 'overlay' and obeys the decomposition axiom.
+-- Complexity: /O((n + m) * log(n))/ time and /O(n + m)/ memory. Note that the
+-- number of edges in the resulting graph is quadratic with respect to the number
+-- of vertices of the arguments: /m = O(m1 + m2 + n1 * n2 + n2 * n1)/.
+--
+-- @
+-- connect x y == connect y x
+-- 'Algebra.Graph.Relation.Symmetric.isEmpty'     (connect x y) == 'Algebra.Graph.Relation.Symmetric.isEmpty'   x   && 'Algebra.Graph.Relation.Symmetric.isEmpty'   y
+-- 'Algebra.Graph.Relation.Symmetric.hasVertex' z (connect x y) == 'Algebra.Graph.Relation.Symmetric.hasVertex' z x || 'Algebra.Graph.Relation.Symmetric.hasVertex' z y
+-- 'Algebra.Graph.Relation.Symmetric.vertexCount' (connect x y) >= 'Algebra.Graph.Relation.Symmetric.vertexCount' x
+-- 'Algebra.Graph.Relation.Symmetric.vertexCount' (connect x y) <= 'Algebra.Graph.Relation.Symmetric.vertexCount' x + 'Algebra.Graph.Relation.Symmetric.vertexCount' y
+-- 'Algebra.Graph.Relation.Symmetric.edgeCount'   (connect x y) >= 'Algebra.Graph.Relation.Symmetric.edgeCount' x
+-- 'Algebra.Graph.Relation.Symmetric.edgeCount'   (connect x y) >= 'Algebra.Graph.Relation.Symmetric.edgeCount' y
+-- 'Algebra.Graph.Relation.Symmetric.edgeCount'   (connect x y) >= 'Algebra.Graph.Relation.Symmetric.vertexCount' x * 'Algebra.Graph.Relation.Symmetric.vertexCount' y `div` 2
+-- 'Algebra.Graph.Relation.Symmetric.vertexCount' (connect 1 2) == 2
+-- 'Algebra.Graph.Relation.Symmetric.edgeCount'   (connect 1 2) == 1
+-- @
+connect :: Ord a => SymmetricRelation a -> SymmetricRelation a -> SymmetricRelation a
+connect (SymmetricRelation x) (SymmetricRelation y) = SymmetricRelation $ Relation 
+                                                      (domain x `Set.union` domain y)
+                                                      (relation x `Set.union` relation y 
+                                                      `Set.union` 
+                                                      (domain x `setProduct` domain y)
+                                                      `Set.union` 
+                                                      (domain y `setProduct` domain x))
+
+instance (Ord a, Num a) => Num (SymmetricRelation a) where
+    fromInteger = vertex . fromInteger
+    x + y       = x `overlay` y
+    x * y       = x `connect` y
+    signum      = const empty
+    abs         = id
+    negate      = id
+
+-- | Check if the internal representation of a relation is consistent, i.e. if all
+-- pairs of elements in the 'relation' refer to existing elements in the 'domain' and that if all pair of edges have their symmetric.
```

Comment:
"all pair of edges have their symmetric" => "all pairs of edges have their symmetric counterparts"